scanner/services/common.go

```diff
@@ -18,8 +18,8 @@ func getClairIndexReport(ctx context.Context, indexer indexer.ReportGetter, hash
 	if !found {
 		return nil, errox.NotFound.Newf("report %q not found", hashID)
 	}
-	if !ir.Success {
-		return nil, errox.NotFound.Newf("unsuccessful state %q: %s", ir.State, ir.Err)
+	if !ir.Success || ir.Err != "" {
```

Comment:
I think it's possible for Success = true but Err != "". I believe it'd be here: https://github.com/quay/claircore/blob/main/indexer/controller/indexfinished.go

Reply:
actually it would be reset to `false` over here https://github.com/quay/claircore/blob/main/indexer/controller/controller.go#L108 but I figure why not be extra safe and just keep this check

Reply:
I don't think it hurts to be extra cautious in case there was something I missed in ClairCore or a bug is introduced in the future where it may accidentally report success when there was an error

Reply:
The same logic follows for "Err" being set to `!= "temporary error in Indexing"` but `Success = true`, maybe due to a bug as you mentioned, and we will have to follow up here. But, I am not going to block on this, even though dependencies not always behave the way we want this is confusing.

Reply:
i'll just remove the err != "" checks